cmd/snap-confine/ns-support.c

```diff
@@ -574,6 +574,25 @@ int sc_join_preserved_per_user_ns(struct sc_mount_ns *group,
 	return ESRCH;
 }
 
+static void setup_signals_for_helper(void)
+{
+	/* Ignore the SIGPIPE signal so that we get EPIPE on the read / write
+	 * operations attempting to work with a closed pipe. This ensures that we
+	 * are not killed by the default disposition (terminate) and can return a
+	 * non-signal-death return code to the program invoking snap-confine. */
```

Comment:
this comment should mention that the issue is that systemd interprets SIGPIPE as a clean termination which is problematic for services that will then not be retried

Reply:
this can be done in a follow up

Reply:
I agree in a followup is fine (cc @mvo5 and @zyga). That followup should remove SIGPIPE on read though since it is only sent on write (we now properly handle read via @mvo5's changes to this PR).